Artesian/_Query/BidAskQuery.py

```diff
@@ -3,55 +3,131 @@
+        """ Set the list of marketdata to be queried.
+
+            Args:
+                ids: list of marketdata id's to be queried. E.g.: 100000xxx
+        """
         super()._forMarketData(ids)
         return self
-    def forFilterId(self, filterId):
+    def forFilterId(self, filterId: int) -> _BidAskQuery:
+        """ Sets the list of filtered marketdata id to be queried
+            
+            Args:
+                filterId: list of marketdata filtered by id"""
         super()._forFilterId(filterId)
         return self
-    def inTimeZone(self, tz):
+    def inTimeZone(self, tz: str) -> _BidAskQuery:
+        """ Gets the BidAsk Query in a specific TimeZone in IANA format.
+
+            Args:
+                tz: "UTC","CET","Europe/Istanbul"
+        """
         super()._inTimezone(tz)
         return self
-    def inAbsoluteDateRange(self, start, end):
+    def inAbsoluteDateRange(self, start: str, end: str) -> _BidAskQuery:
+        """ Gets the BidAsk Query in an absolute date range window. 
+            The Absolute Date Range is in ISO8601 format.
+        
+            Args:
+                start: the date start of the range of extracted timeserie, in ISO format. (ex. "2022-01-01")
+                end:  the EXCLUSIVE date end of the range of extracted timeserie, in ISO format. (ex. "2022-01-01")
+        """
         super()._inAbsoluteDateRange(start, end)
         return self
-    def inRelativePeriodRange(self, pStart, pEnd):
+    def inRelativePeriodRange(self, pStart: str, pEnd: str) -> _BidAskQuery:
+        """ Gets the BidAsk Query in a relative period range time window.
+        
+            Args:
+                pStart: the relative period start of the range of extracted timeseries. (ex. "P--3D")
+                pEnd: the relative period end of the range of the extracted timeseries. (ex. "P10D") 
+        """
+
         super()._inRelativePeriodRange(pStart, pEnd)
         return self
-    def inRelativePeriod(self, extractionPeriod):
+    def inRelativePeriod(self, extractionPeriod: str) -> _BidAskQuery:
+        """ Gets the BidAsk Query in a relative period of a time window.
+        
+            Args:
+                extractionPeriod: the relative period of extracted timeseries. (ex. "P5D")
+        """
         super()._inRelativePeriod(extractionPeriod)
         return self
-    def inRelativeInterval(self, relativeInterval):
+    def inRelativeInterval(self, relativeInterval: str) -> _BidAskQuery:
+        """ Gets the Relative Interval considers a specific interval of time window.
+        
+            Args:
+                relativeInterval: the relative interval of extracted timeseries. (ex. "RelativeInterval.ROLLING_WEEK" or "RelativeInterval.ROLLING_MONTH") 
+        """
         super()._inRelativeInterval(relativeInterval)
         return self
-    def forProducts(self, products):
+    def forProducts(self, products: str) -> _BidAskQuery:
```

Comment:
products type hint is not aligned with sample. I think sample is right, so the type is list of str